craft_application/util/repository.py

```diff
@@ -0,0 +1,59 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2023 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Package repository related functions."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+from craft_cli import emit
+from craft_parts import (
+    LifecycleManager,
+    ProjectInfo,
+)
+
+
+def install_package_repositories(
+    package_repositories: list[dict[str, Any]] | None,
+    lifecycle_manager: LifecycleManager,
+) -> None:
+    """Install package repositories in the environment."""
+    from craft_archives import repo  # type: ignore[import-untyped]
+
+    if not package_repositories:
+        emit.debug("No package repositories specified, none to install.")
+        return
+
+    refresh_required = repo.install(package_repositories, key_assets=Path("/dev/null"))
+    if refresh_required:
+        emit.progress("Refreshing repositories")
+        lifecycle_manager.refresh_packages_list()
+
+    emit.progress("Package repositories installed")
+
+
+def install_overlay_repositories(overlay_dir: Path, project_info: ProjectInfo) -> None:
+    """Install overlay repositories in the environment."""
+    from craft_archives import repo  # type: ignore[import-untyped]
+
+    if project_info.base != "bare":
```

Comment:
I think we need to remove this check here: the concept of a `bare` base doesn't really exist at the craft-application level
Also because this particular combination (package repositories and overlays) only affects Rockcraft, and even there the check is superfluous because [there's already a validator for this](https://github.com/canonical/rockcraft/blob/main/rockcraft/models/project.py#L312). Probably this check came before the validator.

Reply:
So it always call `install_in_root`?

Reply:
yes, but this is only called if `project_mixin.package_repositories` is not None